### ISSUE TITLE
tsc crash involving typing of 'this' in interface member

### ISSUE THREAD

**Issue by tomb-pss** [Microsoft/TypeScript#21086]
**TypeScript Version:**  2.6.2

**Code**

```ts
interface IBase {
    someMember(this: SomeClass): void;
}

interface IFoo extends IBase {}
interface IBar extends IBase {}

type FooBar = IFoo | IBar;

class SomeClass {
    public someMember() {}
}

namespace SomeClass {
    export function callSomeMember(obj: SomeClass & FooBar) {
        obj.someMember();
    }
}
```

**Expected behavior:**
Compiles successfully

**Actual behavior:**
Crashes compiler with the following out:
```
tsc.js:58005
                throw e;
                ^

TypeError: Cannot read property 'flags' of undefined
    at Object.getCheckFlags (tsc.js:8300:22)
    at getTypeOfSymbol (tsc.js:22746:20)
    at tsc.js:23465:113
    at Object.map (tsc.js:469:29)
    at getUnionSignatures (tsc.js:23465:68)
    at resolveUnionTypeMembers (tsc.js:23493:34)
    at resolveStructuredTypeMembers (tsc.js:23703:21)
    at getSignaturesOfStructuredType (tsc.js:24003:32)
    at getSignaturesOfType (tsc.js:24009:20)
    at resolveCallExpression (tsc.js:31759:34)
```

**Related:**
Tripped on this when I accidentally left a typing for "this" where I didn't want one. Don't try to understand what the code is doing - it's been stripped down A LOT to get to just the part that causes the crash.

**Comment by ahejlsberg** [comment]
This looks to be a duplicate of #20802 which is fixed in TS 2.7. There's no error with the current nightly build.

**Comment by tomb-pss** [comment]
Ah, sorry. Missed that when searching for existing issues.
